src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php

```diff
@@ -116,9 +116,12 @@ public function getMetadataFor($value)
             $metadata->mergeConstraints($this->getMetadataFor($parent->name));
         }
 
-        // Include constraints from all implemented interfaces
+        // Include constraints from all implemented interfaces that have not been processed via parent class yet
         foreach ($metadata->getReflectionClass()->getInterfaces() as $interface) {
-            if ('Symfony\Component\Validator\GroupSequenceProviderInterface' === $interface->name) {
+            if (
+                'Symfony\Component\Validator\GroupSequenceProviderInterface' === $interface->name ||
+                $parent && $parent->implementsInterface($interface->name)
```

Comment:
I think it should work :wink: No matter at which level (starting at the parent then going upwards to the parent of the parent etc.) the interface is implemented this condition will be true.

Or what exactly do you mean?